pkg/skaffold/kubernetes/status/resource/deployment.go

```diff
@@ -132,12 +133,24 @@ func (d *Deployment) CheckStatus(ctx context.Context, cfg kubectl.Config) {
 
 	details := d.cleanupStatus(string(b))
 
-	ae := parseKubectlRolloutError(details, err)
-	if ae.ErrCode == proto.StatusCode_STATUSCHECK_KUBECTL_PID_KILLED {
-		ae.Message = fmt.Sprintf("received Ctrl-C or deployments could not stabilize within %v: %v", d.deadline, err)
-	}
-
+	ae := parseKubectlRolloutError(details, d.deadline, err)
 	d.UpdateStatus(ae)
+	// send event update in check status.
+	event.ResourceStatusCheckEventCompleted(d.String(), ae)
+	eventV2.ResourceStatusCheckEventCompleted(d.String(), sErrors.V2fromV1(ae))
+	// if deployment is successfully rolled out, send pod success event to make sure
+	// all pod are marked as success in V2
+	// See https://github.com/GoogleCloudPlatform/cloud-code-vscode-internal/issues/5277
+	if ae.ErrCode == proto.StatusCode_STATUSCHECK_SUCCESS {
+		for _, pod := range d.pods {
+			eventV2.ResourceStatusCheckEventCompletedMessage(
+				pod.String(),
+				fmt.Sprintf("%s %s: running.\n", tabHeader, pod.String()),
+				protoV2.ActionableErr{ErrCode: proto.StatusCode_STATUSCHECK_SUCCESS},
+			)
+		}
+		return
```

Comment:
Yes, I think this would be the case unless I am confusing something.  From the `fetchPods()` snippet here - https://github.com/GoogleContainerTools/skaffold/blob/main/pkg/skaffold/kubernetes/status/resource/deployment.go#L298-L303: 
```
			case proto.StatusCode_STATUSCHECK_SUCCESS:
				event.ResourceStatusCheckEventCompleted(p.String(), p.ActionableError())
				eventV2.ResourceStatusCheckEventCompletedMessage(
					p.String(),
					fmt.Sprintf("%s running.\n", prefix),
					sErrors.V2fromV1(p.ActionableError()))
```
I believe we also send the same `ResourceStatusCheckEventCompletedMessage` event

Reply:
Yes. I agree there could be duplicate sucessful events depending upon timings. 
e.g. 
1) dep check -> waiting for rollout
2) pod check -> unhealhty
... pod becomes healthy
3) Status check sleeps for 100 ms
4) dep rollout status -> successful. 
skaffold send a resource complete event for pod

however in case where. 
1) dep check -> waiting for rollout
.. pod becomes healthy and running
2) pod check -> returns succes. 
skaffold sends a resource complete event for pod
3)  Status check sleeps for 100 ms
4) dep rollout status -> successful. 
skaffold send a another resource complete event for pod

It should be fine to send 2 successful resource complete event for a pod. It would be a no-op on the IDE side.